catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/builder/JsonModelBuilder.java

```diff
@@ -205,7 +205,44 @@ public JsonModelBuilder beginNilType() {
     verifyTerminalNodeNotInProgress();
     // used for date 'IS EMPTY' ops by the UI
     nodeInProgress = new FilterNodeImpl("IS NULL");
-    return null;
+    return this;
+  }
+
+  @Override
+  public JsonModelBuilder addFunctionType(String functionName, List<Object> args) {
+    verifyResultNotYetRetrieved();
+    verifyTerminalNodeNotInProgress();
+
+    FilterNode funcNode = new FilterNodeImpl("FILTER_FUNCTION", functionName);
+    args.forEach(funcNode::addArg);
+
+    if (depth.isEmpty()) {
+      rootNode = funcNode;
+    } else {
+      depth.peek().add(funcNode);
+    }
+    return this;
+  }
+
+  @Override
+  public JsonModelBuilder addBetweenType(String property, Long lower, Long upper) {
```

Comment:
Should this also support float values? I am able to do a `RANGE <float1> TO <float2>` query in the UI.

Reply:
Yeah nice catch that'll impact JSON serialization I'll throw up a fix.

Reply:
@mojogitoverhere Do you think support for `Long` and `Double` will be sufficient? That should cover all cases right?

Reply:
I think we need `Integer` support too for signed values

Reply:
`Long` should support signed values as well so we should be good 👍

Reply:
Scratch that, `Long` supports signed values already